packages/bitcore-wallet-service/src/lib/emailservice.ts

```diff
@@ -236,24 +246,59 @@ export class EmailService {
     });
   }
 
-  _getDataForTemplate(notification, recipient, cb) {
-    // TODO: Declare these in BWU
+  async _getDataForTemplate(notification, recipient, cb) {
     const UNIT_LABELS = {
       btc: 'BTC',
       bit: 'bits',
       bch: 'BCH',
       eth: 'ETH',
       xrp: 'XRP',
       doge: 'DOGE',
-      ltc: 'LTC'
+      ltc: 'LTC',
+      usdc: 'USDC',
+      usdp: 'USDP',
+      gusd: 'GUSD',
+      busd: 'BUSD',
+      wbtc: 'WBTC',
+      dai: 'DAI',
+      shib: 'SHIB',
+      ape: 'APE',
+      euroc: 'EUROC'
     };
 
     const data = _.cloneDeep(notification.data);
     data.subjectPrefix = _.trim(this.subjectPrefix) + ' ';
     if (data.amount) {
       try {
-        const unit = recipient.unit.toLowerCase();
-        data.amount = Utils.formatAmount(+data.amount, unit) + ' ' + UNIT_LABELS[unit];
+        let unit = recipient.unit.toLowerCase();
+        let label = UNIT_LABELS[unit];
+        let opts = {} as any;
+        if (data.tokenAddress) {
+          const tokenAddress = data.tokenAddress.toLowerCase();
+          if (Constants.TOKEN_OPTS[tokenAddress]) {
+            unit = Constants.TOKEN_OPTS[tokenAddress].symbol.toLowerCase();
+            label = UNIT_LABELS[unit];
+          } else {
+            let customTokensData;
+            try {
+              customTokensData = await this.getTokenData();
+            } catch (error) {
+              throw new Error('Could not get custom tokens data');
+            }
+            if (customTokensData && customTokensData[tokenAddress]) {
+              unit = customTokensData[tokenAddress].symbol.toLowerCase();
+              label = unit.toUpperCase();
+              opts.toSatoshis = 10 ** customTokensData[tokenAddress].decimals;
+              opts.decimals = {
+                maxDecimals: 6,
+                minDecimals: 2
+              };
+            } else {
+              throw new Error('Notifications for unsupported token are not allowed');
```

Comment:
same as above, return the cb;

Also it would be good to include the token in the error message, like `'Notifications for unsupported tokens are not allowed: ' + tokenAddress`